jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/params/BoosterParams.scala

```diff
@@ -50,10 +50,21 @@ private[spark] trait BoosterParams extends Params {
    * overfitting. [default=6] range: [1, Int.MaxValue]
    */
   final val maxDepth = new IntParam(this, "maxDepth", "maximum depth of a tree, increase this " +
-    "value will make model more complex/likely to be overfitting.", (value: Int) => value >= 1)
+    "value will make model more complex/likely to be overfitting.", (value: Int) => value >= 0)
```

Comment:
@yanboliang please also take a look at this...I am setting this as 0 to allow users to only set max_leaves in fast histogram algorithm

in approx tree method where 0 is not a valid value, it will crash in native layer as I add a CHECK_EQ assert there

I am not 100%  comfortable with this delegating to native layer....any suggestion?

Reply:
0 should also be valid for distributed `gpu_hist` tree method, as it also supports the `lossguide` grow policy.

Reply:
I don’t think gpu has distributed support and there is no official support of gpu resources allocation in spark

Reply:
There are PRs in flight to add the distributed support based on NCCL2. As for Spark, on Kubernetes there is now pod template support so that GPUs can be allocated for each executor.

Reply:
How it’s related to this PR?

Reply:
0 is support by other algorithms, not just a special case.

Reply:
I don’t understand what you are saying and maybe you need to reword your post

This line in this PR is to support fast histogram algorithm in distributed case which support depth as 0 and delegate checking to native part of approx algorithm

Reply:
All I'm saying is setting `max_depth` to 0 is valid for other algorithms, not just the distributed fast histogram algorithm in this PR. The code as it stands now is fine.

Perhaps you can add a check that if the `tree_method` is `approx`, don't allow `max_depth` to be 0.